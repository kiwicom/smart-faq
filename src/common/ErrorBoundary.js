// @flow

import * as React from 'react';

import ErrorMessage from './ErrorMessage';

const initialState = {
  error: false,
};

type Props = {
  children: React.Node,
};

export type State = {
  error: boolean,
};

class ErrorBoundaryProvider extends React.Component<Props, State> {
  state = initialState;
  componentDidCatch() {
    this.setState({
      error: true,
    });
  }

  render() {
    const { error } = this.state;

    return (
      <React.Fragment>
        {error ? <ErrorMessage /> : this.props.children}
      </React.Fragment>
    );
  }
}

export default ErrorBoundaryProvider;
