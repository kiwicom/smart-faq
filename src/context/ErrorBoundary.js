// @flow

import * as React from 'react';

const initialState = {
  error: false,
  errorDetails: {},
};

type Props = {
  children: React.Node,
};

export type State = {
  error: boolean,
  errorDetails: {},
};

export const ErrorBoundary = React.createContext({
  ...initialState,
});

class ErrorBoundaryProvider extends React.Component<Props, State> {
  state = initialState;
  componentDidCatch(error: Error, errorDetails: {}) {
    this.setState({
      error: true,
      errorDetails,
    });
  }

  render() {
    const { error, errorDetails } = this.state;

    return (
      <ErrorBoundary.Provider
        value={{
          error,
          errorDetails,
        }}
      >
        {this.props.children}
      </ErrorBoundary.Provider>
    );
  }
}

export default ErrorBoundaryProvider;
