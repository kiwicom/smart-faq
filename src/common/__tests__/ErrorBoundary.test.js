// @flow
import * as React from 'react';
import { mount } from 'enzyme';

import ErrorBoundary from '../ErrorBoundary';
import ComponentWithError from './helpers/ComponentWithError.ignore';

describe('ErrorBoundary', () => {
  it('should render normallu', () => {
    const result = mount(
      <ErrorBoundary>
        <h1>This is not an error</h1>
      </ErrorBoundary>,
    );
    expect(result).toMatchSnapshot();
  });
  it('should trow an error', () => {
    const result = mount(
      <ErrorBoundary>
        <ComponentWithError />
      </ErrorBoundary>,
    );
    expect(result).toMatchSnapshot();
  });
});
