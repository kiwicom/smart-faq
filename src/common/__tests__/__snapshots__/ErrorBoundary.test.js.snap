// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorBoundary should render normallu 1`] = `
<ErrorBoundaryProvider>
  <h1>
    This is not an error
  </h1>
</ErrorBoundaryProvider>
`;

exports[`ErrorBoundary should trow an error 1`] = `
<ErrorBoundaryProvider>
  <ErrorMessage>
    <div
      className="jsx-2885065558 errorContainer"
    >
      <div
        className="jsx-2885065558"
      >
        <Heading
          element="h1"
          size="large"
          weight="bold"
        >
          <h1
            className="jsx-2957264124 "
          >
            Something's not quite right.
          </h1>
          <JSXStyle
            css=".__jsx-style-dynamic-selector{font-family:\\"Roboto\\", -apple-system, \\".SFNSText-Regular\\", \\"San Francisco\\", \\"Segoe UI\\", \\"Helvetica Neue\\", \\"Lucida Grande\\", sans-serif;font-size:28px;font-weight:700;color:#171b1e;line-height:1.2;}"
            dynamic={
              Array [
                "\\"Roboto\\", -apple-system, \\".SFNSText-Regular\\", \\"San Francisco\\", \\"Segoe UI\\", \\"Helvetica Neue\\", \\"Lucida Grande\\", sans-serif",
                "28px",
                "700",
                "#171b1e",
                "1.2",
              ]
            }
            styleId="3067881581"
          />
        </Heading>
        <div
          className="jsx-2885065558 desc"
        >
          <Typography
            size="large"
            type="secondary"
          >
            <span
              className="jsx-1855423850"
            >
              Please, try refreshing the page. Sign out. Then sign in again. If the problem persists, don't hesitate to contact us.
              <JSXStyle
                css="span.__jsx-style-dynamic-selector{font-family:Roboto,-apple-system,sans-serif;font-size:16px;color:#7f91a8;;}"
                dynamic={
                  Array [
                    "16px",
                    "#7f91a8",
                    "",
                  ]
                }
                styleId="1112443529"
              />
            </span>
          </Typography>
        </div>
        <withLanguageHOC
          text="Go to contact page"
          textColor="#00a991"
        >
          <ContactPageLink
            language="en"
            text="Go to contact page"
            textColor="#00a991"
          >
            <div
              className="jsx-4293162489 contactUs"
            >
              <a
                className="jsx-4293162489"
                href="/en/content/feedback"
                target="_blank"
              >
                Go to contact page
                <span
                  className="jsx-4293162489 open-icon"
                >
                  <NewWindow
                    color="attention"
                    customColor="#00a991"
                    size="small"
                  >
                    <OrbitIcon
                      color="attention"
                      customColor="#00a991"
                      size="small"
                      viewBox="0 0 24 24"
                    >
                      <IconBase
                        color="#00a991"
                        size="16px"
                        viewBox="0 0 24 24"
                      >
                        <svg
                          fill="currentColor"
                          height="16px"
                          preserveAspectRatio="xMidYMid meet"
                          style={
                            Object {
                              "color": "#00a991",
                              "verticalAlign": "middle",
                            }
                          }
                          viewBox="0 0 24 24"
                          width="16px"
                        >
                          <path
                            d="M19 19H5V5h7V3H5a2 2 0 0 0-2 2v14a2 2 0 0 0 2 2h14c1.1 0 2-.9 2-2v-7h-2v7zM14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3h-7z"
                          />
                        </svg>
                      </IconBase>
                    </OrbitIcon>
                  </NewWindow>
                </span>
              </a>
              <JSXStyle
                css=".contactUs.__jsx-style-dynamic-selector{text-align:center;margin-top:25px;}a.__jsx-style-dynamic-selector{color:#00a991;font-size:14px;-webkit-text-decoration:none;text-decoration:none;font-weight:bold;}a.__jsx-style-dynamic-selector .open-icon.__jsx-style-dynamic-selector{margin-left:6px;}"
                dynamic={
                  Array [
                    "#00a991",
                  ]
                }
                styleId="1280600690"
              />
            </div>
          </ContactPageLink>
        </withLanguageHOC>
      </div>
      <JSXStyle
        css=".errorContainer.jsx-2885065558{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex:1;-ms-flex:1;flex:1;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;width:480px;height:100%;overflow-y:auto;overflow-x:hidden;padding:0 50px;}.desc.jsx-2885065558{margin:10px 0;}"
        styleId="2885065558"
      />
    </div>
  </ErrorMessage>
</ErrorBoundaryProvider>
`;
